src/main/java/is/ru/tictactoe/Board.java

```diff
@@ -0,0 +1,84 @@
+package tictactoe.src.main.java.is.ru.tictactoe;
+
+public class Board {
+	
+	private final static int BOARD_SIZE = 3;
+	private final static int MAX_PLAYERS = 2;
+	private int[][] board;
+	private int moves;
+	
+	public Board() {
+		this.board = new int[BOARD_SIZE][BOARD_SIZE];
+		reset();
+	}
+	
+	private void reset() {
+		this.moves = 0;
+		for (int x = 0; x < BOARD_SIZE; x++) {
+			for (int y = 0; y < BOARD_SIZE; y++) {
+				this.board[x][y] = 0;
+			}
+		}
+	}
+	
+	private void validCell(int x, int y) {
+		if (0 > x || x >= BOARD_SIZE)	throw new java.lang.IndexOutOfBoundsException("Illegal move");
+		if (0 > y || y >= BOARD_SIZE)	throw new java.lang.IndexOutOfBoundsException("Illegal move");		
+	}
+	
+	private void validPlayer(int z) {
```

Comment:
This is a check to see if valid player number is passed to the Board API.
Throws exception if player number is out of scope:

`if (z < 0 || z >= MAX_PLAYERS)  throw new java.lang.IllegalArgumentException("Illegal player");`